File: backend/api/src/main/scala/com/github/ai/leetcodequiz/api/QuestionnaireStatsDto.scala
package com.github.ai.leetcodequiz.api

import zio.json.{JsonDecoder, JsonEncoder}

case class QuestionnaireStatsDto(
  totalQuestions: Int,
  answered: Int,
  notAnswered: Int,
  answeredPositively: Int,
  answeredNegatively: Int
) derives JsonEncoder,
      JsonDecoder

File: android/backend-api/src/main/kotlin/com/github/ai/leetcodequiz/api/QuestionnaireStatsDto.kt
package com.github.ai.leetcodequiz.api

import kotlinx.serialization.Serializable

@Serializable
data class QuestionnaireStatsDto(
    val totalQuestions: Int,
    val answered: Int,
    val notAnswered: Int,
    val answeredPositively: Int,
    val answeredNegatively: Int
)